jenkins/data-prepper/Jenkinsfile

```diff
@@ -0,0 +1,49 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'BRANCH', description: 'Can be: <branchName>, <commitId>, refs/tags/<tagName>', trim: true)
+    }
+    stages {
+        stage('Build Data Prepper') {
+            steps {
+                cleanWs()
+                checkout changelog: false, poll: false, scm:
+                    [$class: 'GitSCM', branches: [[name: BRANCH]], userRemoteConfigs: [[url: 'https://github.com/opensearch-project/data-prepper.git']]]
+                script {
+                    sh './gradlew clean :release:archives:linux:linuxTar -Prelease'
+                }
+            }
+        }
+        stage('Sign Artifacts') {
+            steps {
+                script {
+                    signArtifacts(
+                        artifactPath: "$WORKSPACE/release/archives/linux/build/distributions/",
+                        signatureType: '.sig',
+                        distributionPlatform: 'linux'
+                    )
+                }
+            }
+        }
+        stage('Upload Artifacts to Staging Distribution Bucket') {
+            steps {
+                script {
+                    uploadToS3(
+                        sourcePath: "$WORKSPACE/release/archives/linux/build/distributions/",
+                        bucket: "${ARTIFACT_BUCKET_NAME}",
+                        path: "data-prepper/${VERSION}/${BUILD_NUMBER}/builds/signed"
```

Comment:
I used this path so that multiple staging builds can be produced.

But, for the final release path, I think something like the following would be cleaner:

```
data-prepper/${VERSION}
```

Will this be a problem?

Reply:
> I used this path so that multiple staging builds can be produced.

Let's use the job name in the beginning. It would make it easy to identify which job created this hierarchy. We can have data-prepper in the job name in staging.

Reply:
> But, for the final release path, I think something like the following would be cleaner:
   ```
   data-prepper/${VERSION}
   ```
 >  Will this be a problem?

I believe that works. Promotion to prod should be a one time activity anyways. @peternied - Thoughts on the release path on production?

Reply:
I prefer using job name in case cloned version of the job is run - it prevents generating builds on-top of one another.  Note; the chances of this for a particular component that isn't opensearch/opensearch-dashboards are pretty low.  Up to you @dlvenable